### PR TITLE
feat(skymp5-server): added more edible items

### PR DESCRIPTION
**Pull request by d0gge**
Added more edible items as was requested. Also I decided to move sweetpie edibles logic to `PieScript.cpp` because it started getting messy in `MpActor.cpp`.

### CODE REVIEW

File: skymp5-server/cpp/server_guest_lib/PieScript.cpp [hunk 1]
```diff
@@ -231,14 +230,29 @@ PieScript::PieScript(const std::vector<std::string>& espmFiles)
   AddDLCItems(espmFiles, armorTier5, LootboxItemType::Armor, Tier::Tier5);
   AddDLCItems(espmFiles, consumableTier3, LootboxItemType::Consumable,
               Tier::Tier3);
+
+  miscLootTable = {
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PieScript.cpp:234]: Is it a field?

File: skymp5-server/cpp/server_guest_lib/PieScript.cpp [hunk 2]
```diff
@@ -2,16 +2,15 @@
...
-#include <vector>
-
+#include <stdexcept> #include <unordered_map> #include <vector>
 std::mt19937 g_rng{ std::random_device{}() };
 
-int GenerateRandomNumber(int leftBound, int rightBound)
+uint32_t GenerateRandomNumber(uint32_t leftBound, uint32_t rightBound)
 {
   if (leftBound <= rightBound) {
     std::uniform_int_distribution<> distr(leftBound, rightBound);
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PieScript.cpp:16]: This probably causes implicit uint -> int conversion. Probably not going to cause problems though, but still... I guess that's ok if it was being converted before

File: skymp5-server/cpp/server_guest_lib/PieScript.h [hunk 3]
```diff
@@ -10,7 +10,15 @@ class WorldState;
...
 {
-public:
+private:
+  enum StarterKitChance
+  {
+    ChefKitChance = 25,
+    MinerKitChance = 25,
+    PrisonerKitChance = 25,
+    LumberjackKitChance = 25
+  };
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PieScript.h:20]:
```c++
constexpr static unit32_t kChefKitChance = 25;
...
```

File: skymp5-server/cpp/server_guest_lib/PieScript.h [hunk 4]
```diff
@@ -37,50 +45,49 @@ class PieScript
...
+  LootboxTable lootboxTable;
   std::unordered_map<StarterKitType, std::vector<uint32_t>> starterKitsMap;
+  std::unordered_map<uint32_t, std::vector<uint32_t>> miscLootTable;
 
-  const int TIER1_CHANCE = 600;
-  const int TIER2_CHANCE = 220;
-  const int TIER3_CHANCE = 150;
-  const int TIER4_CHANCE = 25;
-  const int TIER5_CHANCE = 5;
+  const uint32_t TIER1_CHANCE = 600;
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PieScript.h:88]: `kTier1Chance`

File: skymp5-server/cpp/server_guest_lib/PieScript.cpp (hunk 1 above)
Comment (d0gge) [reply skymp5-server/cpp/server_guest_lib/PieScript.cpp:234]: Yep

**Review by nic11 — APPROVED** [review]

**Pull request merged**